sdk/search/Azure.Search.Documents/tests/Utilities/SearchExtensionsTests.cs

```diff
@@ -18,5 +18,40 @@ public void CommaJoin(IEnumerable<string> source, string expected)
             string actual = source.CommaJoin();
             Assert.AreEqual(expected, actual);
         }
+
+        [TestCase(null, new string[] { })]
+        [TestCase("a", new[] { "a" })]
+        [TestCase("a,b", new[] { "a", "b" })]
+        [TestCase("a, b", new[] { "a", " b" })]
+        [TestCase(",b", new[] { "", "b" })]
+        [TestCase("a,b,c", new[] { "a", "b", "c" })]
+        public void CommaSplitBasic(string source, string[] expected)
+        {
+            CollectionAssert.AreEqual(expected, SearchExtensions.CommaSplit(source));
+        }
+
+        [TestCase("BaseRate asc", new[] { "BaseRate asc" })]
+        [TestCase("Rating desc,BaseRate", new[] { "Rating desc", "BaseRate" })]
+        [TestCase(
+            "Rating desc,geo.distance(Location, geography'POINT(-122.131577 47.678581)') asc",
+            new[]
+            {
+                "Rating desc",
+                "geo.distance(Location, geography'POINT(-122.131577 47.678581)') asc"
+            })]
+        [TestCase(
+            "search.score() desc,Rating desc,geo.distance(Location, geography'POINT(-122.131577 47.678581)') asc",
+            new[]
+            {
+                "search.score() desc",
+                "Rating desc",
+                "geo.distance(Location, geography'POINT(-122.131577 47.678581)') asc"
+            })]
+        [TestCase("'inside literal , ( )'','", new[] { "'inside literal , ( )'','" })]
```

Comment:
I like this test case, but would love to see it modified (or one added) to have this and another, possibly even simple clause just to make sure significant commas are still split.